### PR TITLE
Fixed Issue 15800: std.conv.to!int does not work with ranges of any char …

### PR DESCRIPTION
**Pull request by JackStouffer**
…type

### CODE REVIEW

File: std/conv.d [hunk 1]
```diff
@@ -1728,19 +1728,19 @@ private void testFloatingToIntegral(Floating, Integral)()
 
 
 /**
-String to non-string conversion runs parsing.
+String, or string like input range, to non-string conversion runs parsing.
```
Comment (andralex) [inline std/conv.d]: string-like

File: std/conv.d [hunk 2]
```diff
@@ -1728,19 +1728,19 @@ private void testFloatingToIntegral(Floating, Integral)()
...
 $(UL
   $(LI When the source is a wide string, it is first converted to a narrow
        string and then parsed.)
   $(LI When the source is a narrow string, normal text parsing occurs.))
 */
 private T toImpl(T, S)(S value)
-    if ( isExactSomeString!S && isDynamicArray!S &&
```
Comment (andralex) [inline std/conv.d]: When did this `isExactSomeString` come about? Sigh...

**Comment by d-random-contributor** [comment]
Is the autotester working?

**Comment by andralex** [comment]
lgtm modulo the doc nit

**Comment by JackStouffer** [comment]
@andralex fixed

**Comment by schveiguy** [comment]
Auto-merge toggled on

**Comment by JackStouffer** [comment]
Thanks!

**Pull request merged**